src/main/java/duke/Storage.java

```diff
@@ -0,0 +1,60 @@
+package duke;
+
+import duke.exception.LoadFileFailDukeException;
+import duke.exception.WriteFileFailDukeException;
+import place.Place;
+
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.ObjectInputStream;
+import java.io.ObjectOutputStream;
+import java.util.ArrayList;
+import java.util.HashMap;
+
+public class Storage {
+    private String filePath;
+
+    /**
+     * Initialise filePath for continuous referencing.
+     *
+     * @param filePath filePath to save file
+     */
+    public Storage(String filePath) {
+        this.filePath = filePath;
+    }
+
+
+    /**
+     * Try to load raw data from filePath and deserialize everything.
+     *
+     * @throws LoadFileFailDukeException if file or raw data can't be loaded
+     */
+    @SuppressWarnings("unchecked")
+    public TaskList load() throws LoadFileFailDukeException {
+        try {
+            ObjectInputStream ois = new ObjectInputStream(new FileInputStream(filePath));
+            Place.setRefLatLong((HashMap) ois.readObject());
+            Place.setRefAlias((HashMap) ois.readObject());
+            return (TaskList) ois.readObject();
+        } catch (Exception e) {
```

Comment:
I don't think it's a good idea to catch a general exception. For example, if it is a FileNotFoundException, a new file could be created with a clearer error message printed.

Reply:
👍 Agreed. Scope should always be minimal.